### ISSUE TITLE
.dialog file debugging is broken

### ISSUE THREAD

**Issue by xieofxie** [microsoft/botbuilder-dotnet#3895]
## Version
What package version of the SDK are you using.

Current master

## Describe the bug
Give a clear and concise description of what the bug is.

.dialog file debugging is broken

## To Reproduce
1. install https://marketplace.visualstudio.com/items?itemName=tomlm.vscode-dialog-debugger
2. run Testbot.Json
3. attach in vscode and breakpoints are not working

## Expected behavior
Give a clear and concise description of what you expected to happen.

## Additional context

Part of the issue was caused by this https://github.com/microsoft/botbuilder-dotnet/pull/3857
`var pass1Result = token.ToObject<T>(serializer);` registers source code twice and makes breakpoints can't match to dialogs.

[bug]

**Comment by xieofxie** [comment]
It seems some blocks are added both as file and function.

A workaround: replace https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.Dialogs.Debugging/DebuggerSourceMap.cs#L259 with
```
            var opts = options.ToArray();
            bool result = TryUpdate(row, opts[0]);
            if (opts.Length >= 2)
            {
                result |= TryUpdate(row, opts[1]);
            }

            return result;
```

**Comment by gabog** [comment]
Hi @xieofxie, can you confirm this was closed by the related PR? If yes, please close this issue too.

**Comment by xieofxie** [comment]
@gabog No, since it is not migrated to current master 03a51e1e09d36cd8c4e8f1dcde6263e5d62e806e (https://github.com/microsoft/botbuilder-dotnet/issues/3970), this issue is not resolved yet.

**Comment by gabog** [comment]
Ah, OK thanks.

**Comment by cleemullins** [comment]
We believe this was fixed by @tomlm merge of 4.9 into master last week. Please confirm.

**Comment by xieofxie** [comment]
Yes, it worked now. Hope we could find some way to UT this..
